src/main/java/com/librato/disco/DiscoClient.java

```diff
@@ -124,10 +124,10 @@ public void stop() throws IOException {
 
     Node<T> _toNode(ChildData data) {
         String path = pathFromData(data);
-        String[] split = path.split(":");
-        if (split.length != 2) {
-            throw new RuntimeException("Don't know how to parse node path: " + path);
-        }
+        // This is somewhat hacky support for ipv6 with the same host:port notation
+        int l = path.lastIndexOf(':');
```

Comment:
If we remove the requirement for the host and port to be in the path i think we can handle that then. Otherwise I'm okay to let it fail.